lib/erbse/parser.rb

```diff
@@ -4,6 +4,7 @@ class Parser
     ERB_EXPR = /<%(=+|-|\#|@\s|%)?(.*?)[-=]?%>(\n)*/m # this is for backward-compatibility.
     # BLOCK_EXPR     = /\s*((\s+|\))do|\{)(\s*\|[^|]*\|)?\s*\Z/
     BLOCK_EXPR = /\b(if|unless)\b|\sdo\s*$|\sdo\s+\|/
```

Comment:
So you basically simply removed the `if|unless` pattern? And it still works? Ingenious!

Reply:
Hi @apotonick,
maybe you've overseen it. I removed the `if|unless` pattern only in the `BLOCK_EXPR_OUT` constant that is used only in expression tags `<%= ... %>`. So there the keywords `if` and `else` should be interpreted as inline or postfix condition.
But in execution tags `<% ... %>` I want to have the old behaviour and interprete `if|unless` as block expressions. So I use the old constant `BLOCK_EXPR` there with `if|unless`.

The only problem I see with this solution is this one:
```
<% foo = 'bar' if true %>
<%= foo %>
```
So maybe I find another way of solving this.

Reply:
Can you add a breaking test for this, please? We can then discuss the solution.